### DISCUSSION TITLE
Reading edge list from csv line

### DISCUSSION THREAD

**Discussion by vadhoob** [krishnanlab/PecanPy#129] [Q&A] [answered]
Hi, 

I am new to graph ML and have been trying to play with node2vec for the past couple of days. I have tried several implementations of which and they all seem quite slow when it comes to generating the baised walk. I am keen to use the pecanpy implementation however my edge list is in a csv file (attached) and I am not clear how to convert it to .edg format pecanpy is expecting it to be. 

I have tried looking around but haven't found anything that works. Any help would be greatly appreciated! 

Cheers! 

[musae_git_edges.csv](https://github.com/krishnanlab/PecanPy/files/9084967/musae_git_edges.csv)

**Comment by RemyLau** [comment, marked as answer]
Hi @vadhoob, thanks for your interest in using PecanPy! A recent patch #66 made by @justinlboyer  allows one to specify the delimiter type. So essentially, you can pass your csv file directly to the PecanPy program by specifying `--delimiter ,`.

One note I want to make is that the current implementation can not exclude the harder line automatically. So you will need to manually remove the header before running PecanPy, either using a text editor or using the following command

```bash
$ tail -n +2 csv_with_header.csv > csv_without_header.csv
```

Finally, here's an example script you can use the run PecanPy with your csv file

```bash
$ pecanpy --input csv_without_header.csv --delimiter , --mode SparseOTF --output out.emd
```

**Reply by vadhoob** [reply]
thanks @RemyLau that worked! I am impressed at how quick it runs! 

For anyone using the API, after removing the header, this is what I did: 

`g.read_edg('_edges_without_headers.csv',
           delimiter=',',
           weighted=False, directed=False)
`
My file was separated by ',' . 

Cheers!

**Reply by RemyLau** [reply]
Sounds great! I'm glad that it worked out for you!
